rfcs/0000-poa-testnet/0000-poa-testnet.md

```diff
@@ -0,0 +1,124 @@
+* If the validator is the attester of the last block, wait for `ATTEST_INTERVAL` blocks then continue this strategy.
+
+`ATTEST_INTERVAL` is used for preventing malicious validators to censors the POA network. When malicious validators are less than `ATTEST_INTERVAL + 1` that at least an honest validator have the chance to submit votes and evict malicious validators.
+
+`ATTEST_INTERVAL` can be set to `VALIDATOR_COUNT / 2` the honest validators could eventually evict malicious validators unless the half of validators are corrupted.
+
+## validator list
+
+Validator list can be represented as a list of pubkey hash.
+
+``` txt
+pubkey_hash_1 | pubkey_hash_2 | pubkey_hash_3 ...
+```
+
+We can use an `M of N` multisig lock to describe the validator list, the `M` is set to `VOTE_LIMIT`, and `N` is `VALIDATORS_COUNT`.
+
+To change the validator list, a validator can collect enough votes(the signatures) and send a tx to spent the old multisig lock and construct a new multisig lock with new pubkey hashes. A validator node can use `type_id` to track this validator list.
+
+The pre-defined [multisig script](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0021-ckb-address-format/0021-ckb-address-format.md#short-payload-format) is satisfied the requirements, one except is the `multisig script` should be public so other nodes can check new validator list. Validators should make sure the entire `multisig script` is stored to cell data, and only one cell is constructed to represent the validator list.
+
+Notice, the process of collect data is not included in the POA protocol for simplifying, a responsible validator should check the vote tx carefully before signing it.
+
+## POA header
+
+Unfortunately, CKB header is fixed-length, the `nonce` field takes 128 bits, we can't just put 256 bits secp256k1 signature into the header.
+
+For easier implementation, instead change the block header structure, we put an extra POA context concatenated after cellbase transaction's first witness. To verify a header attestation, we need both the header and the cellbase transaction.
+
+The structure of `POAContext`:
+
+``` txt
+table POAContext {
+    signature:              Byte65,
+    transactions_count:     Uint32,
+    raw_transactions_root:  Byte32, // witness transactions root
+    merkle_proof:           Byte32Vec, // merkle proof of cellbase and voting txs
+    voting_txs:             TransactionVec,
+}
+```
+
+The first field `signature` is a secp256k1 signature signed by a validator.
+
+The signature message is supposed to digest the whole block except the signature itself.
+
+Because the signature is put into the cellbase transaction's witness, and the cellbase itself is digested by the `transaction_root` in the header. If we change the cellbase transaction the `transaction_root` and block hash will also be changed.
```

Comment:
```suggestion
Because the signature is put into the cellbase transaction's witness, and the cellbase itself is digested by the `transaction_root` in the header. If we change the cellbase transaction the `transaction_root` and block hash will also be changed. A validator needs to 'zeroize' the signature field before signing.
```